From 192b8c0b93e2755518cd04ef1197a774dc8083bd Mon Sep 17 00:00:00 2001
From: Gilles Dartiguelongue <eva@gentoo.org>
Date: Wed, 26 Jan 2011 21:40:14 +0100
Subject: [PATCH] Fix tests link failures

It looks like internal copy of libqof needs to be explicitely mentionned
in a couple of places for tests to find the proper library and not the
one already installed on the system which may or may not have the
required symbols.
---
 src/app-utils/test/Makefile.am                |    2 +-
 src/backend/xml/test/Makefile.am              |    5 +++--
 src/bin/Makefile.am                           |   10 +++++-----
 src/business/business-core/test/Makefile.am   |    2 +-
 src/core-utils/test/Makefile.am               |    2 +-
 src/engine/test/Makefile.am                   |    6 +++---
 src/import-export/ofx/test/Makefile.am        |    1 +
 src/import-export/qif-import/test/Makefile.am |    2 +-
 src/import-export/test/Makefile.am            |    3 ++-
 src/register/ledger-core/test/Makefile.am     |    1 +
 src/register/register-core/test/Makefile.am   |    3 ++-
 src/register/register-gnome/test/Makefile.am  |    1 +
 src/report/report-gnome/test/Makefile.am      |    1 +
 13 files changed, 23 insertions(+), 16 deletions(-)

diff --git a/src/app-utils/test/Makefile.am b/src/app-utils/test/Makefile.am
index 52baf4a..5c0fd96 100644
--- a/src/app-utils/test/Makefile.am
+++ b/src/app-utils/test/Makefile.am
@@ -33,13 +33,13 @@ TESTS_ENVIRONMENT = \
   $(shell ${top_srcdir}/src/gnc-test-env --no-exports ${GNC_TEST_DEPS})
 
 LDADD = \
+   ${top_builddir}/src/libqof/qof/libgnc-qof.la \
+   ${top_builddir}/src/core-utils/libgnc-core-utils.la \
    ${top_builddir}/src/engine/libgncmod-engine.la \
    ${top_builddir}/src/gnc-module/libgnc-module.la \
    ${top_builddir}/src/app-utils/libgncmod-app-utils.la \
    ${top_builddir}/src/test-core/libtest-core.la \
    ${top_builddir}/src/engine/test-core/libgncmod-test-engine.la \
-   ${top_builddir}/src/core-utils/libgnc-core-utils.la \
-   ${top_builddir}/src/libqof/qof/libgnc-qof.la \
    ${GUILE_LIBS} \
    ${GCONF_LIBS}
 
diff --git a/src/backend/xml/test/Makefile.am b/src/backend/xml/test/Makefile.am
index 85c5f75..8989542 100644
--- a/src/backend/xml/test/Makefile.am
+++ b/src/backend/xml/test/Makefile.am
@@ -218,14 +218,15 @@ check_PROGRAMS = \
 
 noinst_HEADERS = test-file-stuff.h
 
-LDADD = ${top_builddir}/src/test-core/libtest-core.la \
+LDADD = \
+        ${top_builddir}/src/libqof/qof/libgnc-qof.la \
+        ${top_builddir}/src/test-core/libtest-core.la \
         ${top_builddir}/src/gnc-module/libgnc-module.la \
         ${top_builddir}/src/engine/libgncmod-engine.la \
         ${top_builddir}/src/engine/test-core/libgncmod-test-engine.la \
         ${top_builddir}/src/core-utils/libgnc-core-utils.la \
         ./libgnc-test-file-stuff.la \
         ${LIBXML2_LIBS} \
-        ${top_builddir}/src/libqof/qof/libgnc-qof.la \
         ${top_builddir}/lib/libc/libc-missing.la
 
 AM_CPPFLAGS = \
diff --git a/src/bin/Makefile.am b/src/bin/Makefile.am
index f740b39..1e70d98 100644
--- a/src/bin/Makefile.am
+++ b/src/bin/Makefile.am
@@ -61,16 +61,16 @@ bin_PROGRAMS = ${BIN_NAME}
 gnucash_SOURCES = gnucash-bin.c ${GNUCASH_RESOURCE_FILE}
 gnucash_LDADD = \
   ${top_builddir}/src/libqof/qof/libgnc-qof.la \
+  ${top_builddir}/src/core-utils/libgnc-core-utils.la \
-  ${GUILE_LIBS} \
-  ${GLIB_LIBS} \
-  ${GNOME_LIBS} \
-  ${GTK_LIBS} \
   ${top_builddir}/src/engine/libgncmod-engine.la \
   ${top_builddir}/src/gnome/libgnc-gnome.la \
   ${top_builddir}/src/gnome-utils/libgncmod-gnome-utils.la \
   ${top_builddir}/src/app-utils/libgncmod-app-utils.la \
-  ${top_builddir}/src/core-utils/libgnc-core-utils.la \
-  ${top_builddir}/src/gnc-module/libgnc-module.la
+  ${top_builddir}/src/gnc-module/libgnc-module.la \
+  ${GUILE_LIBS} \
+  ${GLIB_LIBS} \
+  ${GNOME_LIBS} \
+  ${GTK_LIBS}
 
 GNUCASH_BIN_INSTALL_NAME=`echo ${BIN_NAME} | sed -e '$(transform)'`
 
diff --git a/src/business/business-core/test/Makefile.am b/src/business/business-core/test/Makefile.am
index 9495d13..d3fec0d 100644
--- a/src/business/business-core/test/Makefile.am
+++ b/src/business/business-core/test/Makefile.am
@@ -11,11 +11,11 @@ AM_CPPFLAGS = \
   ${GLIB_CFLAGS}
 
 LDADD = \
+  ${top_builddir}/src/libqof/qof/libgnc-qof.la \
   ${top_builddir}/src/gnc-module/libgnc-module.la \
   ${top_builddir}/src/test-core/libtest-core.la \
   ${top_builddir}/src/engine/libgncmod-engine.la \
   ../libgncmod-business-core.la \
-  ${top_builddir}/src/libqof/qof/libgnc-qof.la \
   ${GLIB_LIBS}
 
 TESTS = \
diff --git a/src/core-utils/test/Makefile.am b/src/core-utils/test/Makefile.am
index 7b3c131..4bd2759 100644
--- a/src/core-utils/test/Makefile.am
+++ b/src/core-utils/test/Makefile.am
@@ -9,9 +9,9 @@ AM_CPPFLAGS = \
   ${GLIB_CFLAGS}
 
 LDADD = \
+  ${top_builddir}/src/libqof/qof/libgnc-qof.la \
   ../libgnc-core-utils.la \
   ${top_builddir}/src/test-core/libtest-core.la \
-  ${top_builddir}/src/libqof/qof/libgnc-qof.la \
   ${GLIB_LIBS}
 
 # these tests are ordered kind more or less in the order
diff --git a/src/engine/test/Makefile.am b/src/engine/test/Makefile.am
index 0682759..0c9ab68 100644
--- a/src/engine/test/Makefile.am
+++ b/src/engine/test/Makefile.am
@@ -12,12 +12,12 @@ AM_CPPFLAGS = \
   ${GLIB_CFLAGS}
 
 LDADD = \
+  ${top_builddir}/src/libqof/qof/libgnc-qof.la \
   ${top_builddir}/src/gnc-module/libgnc-module.la \
   ${top_builddir}/src/test-core/libtest-core.la \
   ../libgncmod-engine.la \
   ../test-core/libgncmod-test-engine.la \
   ${top_builddir}/src/core-utils/libgnc-core-utils.la \
-  ${top_builddir}/src/libqof/qof/libgnc-qof.la \
   ${GLIB_LIBS}
 
 # these tests are ordered kind more or less in the order
@@ -84,8 +84,8 @@ check_PROGRAMS = \
 
 test_link_SOURCES = test-link.c
 test_link_LDADD = ../libgncmod-engine.la \
-  ${top_builddir}/src/core-utils/libgnc-core-utils.la \
-  ${top_builddir}/src/libqof/qof/libgnc-qof.la
+  ${top_builddir}/src/libqof/qof/libgnc-qof.la \
+  ${top_builddir}/src/core-utils/libgnc-core-utils.la
 
 EXTRA_DIST = \
   test-create-account \
diff --git a/src/import-export/ofx/test/Makefile.am b/src/import-export/ofx/test/Makefile.am
index 0283ed7..81d7f4b 100644
--- a/src/import-export/ofx/test/Makefile.am
+++ b/src/import-export/ofx/test/Makefile.am
@@ -5,5 +5,6 @@ check_PROGRAMS=test-link
 
 test_link_SOURCES=test-link.c
 test_link_LDADD=\
+	$(top_builddir)/src/libqof/qof/libgnc-qof.la \
 	${top_builddir}/src/gnome-utils/libgncmod-gnome-utils.la \
     ../libgncmod-ofx.la 
diff --git a/src/import-export/qif-import/test/Makefile.am b/src/import-export/qif-import/test/Makefile.am
index 118354c..eabcc86 100644
--- a/src/import-export/qif-import/test/Makefile.am
+++ b/src/import-export/qif-import/test/Makefile.am
@@ -6,11 +6,11 @@ test_link_SOURCES = \
   test-link.c
 
 test_link_LDADD=../libgncmod-qif-import.la \
+  ${top_builddir}/src/libqof/qof/libgnc-qof.la \
   ${top_builddir}/src/core-utils/libgnc-core-utils.la \
   ${top_builddir}/src/gnome/libgnc-gnome.la \
   ${top_builddir}/src/engine/libgncmod-engine.la \
   ${top_builddir}/src/gnome-utils/libgncmod-gnome-utils.la \
   ${top_builddir}/src/app-utils/libgncmod-app-utils.la \
-  ${top_builddir}/src/libqof/qof/libgnc-qof.la \
   ${LIBXML2_LIBS}
 
diff --git a/src/import-export/test/Makefile.am b/src/import-export/test/Makefile.am
index f9be395..9ae60ff 100644
--- a/src/import-export/test/Makefile.am
+++ b/src/import-export/test/Makefile.am
@@ -10,12 +10,13 @@ AM_CPPFLAGS = \
   ${GLIB_CFLAGS}
 
 LDADD = \
+  ${top_builddir}/src/libqof/qof/libgnc-qof.la \
+  ${top_builddir}/src/core-utils/libgnc-core-utils.la \
   ${top_builddir}/src/gnc-module/libgnc-module.la \
   ${top_builddir}/src/test-core/libtest-core.la \
   ../libgncmod-generic-import.la \
   ${top_builddir}/src/gnome-utils/libgncmod-gnome-utils.la \
   ${top_builddir}/src/engine/libgncmod-engine.la \
-  ${top_builddir}/src/libqof/qof/libgnc-qof.la \
   ${GLIB_LIBS}
 
 TESTS = \
diff --git a/src/register/ledger-core/test/Makefile.am b/src/register/ledger-core/test/Makefile.am
index f6461cf..ca78cf5 100644
--- a/src/register/ledger-core/test/Makefile.am
+++ b/src/register/ledger-core/test/Makefile.am
@@ -4,6 +4,7 @@ check_PROGRAMS = test-link-module
 
 test_link_module_SOURCES=test-link-module.c
 test_link_module_LDADD=\
+	$(top_builddir)/src/libqof/qof/libgnc-qof.la \
 	${top_builddir}/src/gnome-utils/libgncmod-gnome-utils.la \
     ../libgncmod-ledger-core.la
 
diff --git a/src/register/register-core/test/Makefile.am b/src/register/register-core/test/Makefile.am
index 20a240d..e4c1126 100644
--- a/src/register/register-core/test/Makefile.am
+++ b/src/register/register-core/test/Makefile.am
@@ -12,7 +12,8 @@ check_PROGRAMS = \
   test-link-module
 
 test_link_module_SOURCES=test-link-module.c
-test_link_module_LDADD= \
+test_link_module_LDADD=\
+  $(top_builddir)/src/libqof/qof/libgnc-qof.la \
   ${top_builddir}/src/engine/libgncmod-engine.la \
   ${top_builddir}/src/app-utils/libgncmod-app-utils.la \
   ${top_builddir}/src/gnome-utils/libgncmod-gnome-utils.la \
diff --git a/src/register/register-gnome/test/Makefile.am b/src/register/register-gnome/test/Makefile.am
index 83409d6..ddf3347 100644
--- a/src/register/register-gnome/test/Makefile.am
+++ b/src/register/register-gnome/test/Makefile.am
@@ -13,6 +13,7 @@ check_PROGRAMS = \
 
 test_link_module_SOURCES = test-link-module.c
 test_link_module_LDADD = \
+  $(top_builddir)/src/libqof/qof/libgnc-qof.la \
   ${top_builddir}/src/engine/libgncmod-engine.la \
   ${top_builddir}/src/app-utils/libgncmod-app-utils.la \
   ${top_builddir}/src/gnome-utils/libgncmod-gnome-utils.la \
diff --git a/src/report/report-gnome/test/Makefile.am b/src/report/report-gnome/test/Makefile.am
index 5884ccd..f31c4cf 100644
--- a/src/report/report-gnome/test/Makefile.am
+++ b/src/report/report-gnome/test/Makefile.am
@@ -20,6 +20,7 @@ TESTS_ENVIRONMENT = \
   $(shell ${top_srcdir}/src/gnc-test-env --no-exports ${GNC_TEST_DEPS})
 
 LDADD = \
+	$(top_builddir)/src/libqof/qof/libgnc-qof.la \
     ${top_builddir}/src/report/report-gnome/libgncmod-report-gnome.la \
 	${top_builddir}/src/gnome-utils/libgncmod-gnome-utils.la
 
-- 
1.7.4.rc2

